browser/static/js/adjust.js

```diff
@@ -1,277 +1,290 @@
-        continue;
-      // outline white single
-      } else if (singleOutline && currentVal === -o1) {
-        recolorScaled(ann, i, j, jlen, r=255, g=255, b=255);
-        continue;
-      // outline all remaining edges with white
-      } else if (outlineAll && currentVal < 0) {
-        recolorScaled(ann, i, j, jlen, r=255, g=255, b=255);
-        continue;
-      // translucent highlight
-      } else if (translucent &&
-            (Math.abs(currentVal) === t1 || Math.abs(currentVal) === t2)) {
-        recolorScaled(ann, i, j, jlen, r=60, g=60, b=60);
-        continue;
+  postCompositeLabelMod(img,
+      redOutline=false, r1=-1,
+      singleOutline=false, o1=-1,
+      outlineAll = false,
+      translucent=false, t1=-1, t2=-1) {
+
+    let r, g, b;
+    let ann = img.data;
+    // use label array to figure out which pixels to recolor
+    for (var j = 0; j < seg_array.length; j += 1){ //y
+      for (var i = 0; i < seg_array[j].length; i += 1){ //x
+        let jlen = seg_array[j].length;
+        let currentVal = seg_array[j][i];
+        // outline red
+        if (redOutline && currentVal === -r1) {
+          this._recolorScaled(ann, i, j, jlen, r=255, g=-255, b=-255);
+          continue;
+        // outline white single
+        } else if (singleOutline && currentVal === -o1) {
+          this._recolorScaled(ann, i, j, jlen, r=255, g=255, b=255);
+          continue;
+        // outline all remaining edges with white
+        } else if (outlineAll && currentVal < 0) {
+          this._recolorScaled(ann, i, j, jlen, r=255, g=255, b=255);
+          continue;
+        // translucent highlight
+        } else if (translucent &&
+              (Math.abs(currentVal) === t1 || Math.abs(currentVal) === t2)) {
+          this._recolorScaled(ann, i, j, jlen, r=60, g=60, b=60);
+          continue;
+        }
       }
     }
   }
-}
 
-// apply contrast+brightness to raw image
-function contrastRaw(contrast, brightness) {
-  let canvas = document.getElementById('hidden_seg_canvas');
-  let ctx = $('#hidden_seg_canvas').get(0).getContext('2d');
-  ctx.imageSmoothingEnabled = false;
+  // apply contrast+brightness to raw image
+  contrastRaw(rawImage, contrastedRaw, contrast, brightness) {
 
-  // draw seg_image so we can extract image data
-  ctx.clearRect(0, 0, rawWidth, rawHeight);
-  ctx.drawImage(raw_image, 0, 0, rawWidth, rawHeight);
-  let rawData = ctx.getImageData(0, 0, rawWidth, rawHeight);
-  contrast_image(rawData, contrast, brightness);
-  ctx.putImageData(rawData, 0, 0);
+    // draw rawImage so we can extract image data
+    this.ctx.clearRect(0, 0, this.width, this.height);
+    this.ctx.drawImage(rawImage, 0, 0, this.width, this.height);
+    let rawData = this.ctx.getImageData(0, 0, this.width, this.height);
+    this._contrast_image(rawData, contrast, brightness);
+    this.ctx.putImageData(rawData, 0, 0);
 
-  contrastedRaw.src = canvas.toDataURL();
-}
+    contrastedRaw.src = this.canvas.toDataURL();
```

Comment:
Okay, so more along the lines of
```
preCompAdjust () {
  // function body
return canvas.toDataURL();
}
```
?
And then within the main js code, instead of calling `adjuster.preCompAdjust(preCompSeg);` it's more like
```
preCompSeg.src = adjuster.preCompAdjust();
```

Reply:
Actually, while I think that approach would work for most of the places these functions are called, it won't work as well within `start_caliban`:
`    rawImage.onload = () => adjuster.contrastRaw(rawImage, contrastedRaw, current_contrast, brightness);`
Not sure how to address this to nicely handle both use cases.

Reply:
Would that code not work?

I haven't run anything, but the code you have right there seems like it would work.